biofefi/services/preprocessing.py

```diff
@@ -16,6 +17,35 @@
 from biofefi.services.configuration import save_options
 
 
+def find_non_numeric_columns(data: Union[pd.DataFrame, pd.Series]) -> List[str]:
```

Comment:
```suggestion
def find_non_numeric_columns(data: pd.DataFrame | pd.Series) -> list[str]:
```
In python 3.11+ you can use the `|` on types. And I think you can do square brackets on `list` to add the content type